assets/src/scripts/monitoring-location/components/hydrograph/download-links.js

```diff
@@ -87,43 +58,53 @@ export const renderDownloadLinks = function(elem, store, siteno) {
                 }
         };
 
-        if (hasIVData) {
+        if (hasIVData && primaryIVData && Object.keys(primaryIVData.values) > 0) {
             listOfDownloadLinks.append('li')
                 .call(createDataDownloadLink, {
                     displayText: 'Current IV data',
-                    url: createUrlForDownloadLinks(currentIVDateRange, queryInformation, parameterCode, 'current'),
+                    url: getServiceURL({
+                        monitoringLocations: monitoringLocations,
+                        parameterCode: inputs.parameterCode,
+                        startTime: startDT,
+                        endTime: endDT,
+                        format: 'rdb'
+                    }),
                     gaEventAction: 'downloadLinkCurrent',
                     tooltipText: 'Monitoring location data as shown on graph'
                 });
         }
 
-        if (hasIVData && showIVTimeSeries.compare) {
+        if (inputs.loadCompare && compareIVData && Object.keys(compareIVData.values) > 0) {
```

Comment:
Same suggestion as above.

Reply:
lol. I knew there was another way to do this, but was blanking on it.